cmd/util/cmd/epoch-recovery/cmd/reset.go

```diff
@@ -0,0 +1,207 @@
+
+	// set payout
+	payout := uint64(0)
+
+	// read random source from epoch
+	randomSource, err := epoch.RandomSource()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get random source from epoch")
+	}
+
+	// read first view
+	firstView, err := epoch.FirstView()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get first view from epoch")
+	}
+
+	// read final view
+	finalView, err := epoch.FinalView()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get final view from epoch")
+	}
+
+	// read collector clusters and convert to strings
+	clustering, err := epoch.Clustering()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get clustering from epoch")
+	}
+
+	// read dkg public keys for all participants
+	initialIdentities, err := epoch.InitialIdentities()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get initial identities from epoch")
+	}
+
+	dkg, err := epoch.DKG()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get dkg from epoch")
+	}
+
+	dkgKeys := make([]crypto.PublicKey, 0, len(initialIdentities))
+	for _, identity := range initialIdentities {
+		keyShare, err := dkg.KeyShare(identity.NodeID)
+		if err != nil {
+			log.Fatal().Str("node_id", identity.NodeID.String()).Err(err).Msg("coiuld get key share for node id")
+		}
+		dkgKeys = append(dkgKeys, keyShare)
+	}
+
+	// TODO: read in QCs
+
+	return convertResetEpochArgs(randomSource, payout, firstView, finalView, clustering, []string{}, dkgKeys)
+}
+
+// convertResetEpochArgs converts the arguments required by `resetEpoch` to cadence representations
+func convertResetEpochArgs(randomSource []byte,
+	payout, firstView, finalView uint64,
+	clustering flow.ClusterList, clusterQCs []string, dkgPubKeys []crypto.PublicKey) cadence.Array {
+
+	args := make([]cadence.Value, 0)
+
+	// add random source
+	args = append(args, cadence.NewString(hex.EncodeToString(randomSource)))
+
+	// add payout
+	cdcPayout, err := cadence.NewUFix64(fmt.Sprint(payout))
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not convert payout to cadence type")
+	}
+	args = append(args, cdcPayout)
+
+	// add first view
+	args = append(args, cadence.NewUInt64(firstView))
+
+	// add final view
+	args = append(args, cadence.NewUInt64(finalView))
+
+	// add clusters
+	cdcClusters := make([]cadence.Value, len(clustering))
+	for i, cluster := range clustering {
+		clusterNodeIDs := cluster.NodeIDs()
+		nodeIDStrings := make([]cadence.Value, len(clusterNodeIDs))
+		for j, nodeID := range clusterNodeIDs {
+			nodeIDStrings[j] = cadence.NewString(nodeID.String())
+		}
+		cdcClusters[i] = cadence.NewArray(nodeIDStrings)
+	}
+	args = append(args, cadence.NewArray(cdcClusters))
+
+	// TODO:  clusterQC
+
+	// add dkg keys
+	cdcDKGKeys := make([]cadence.Value, len(dkgPubKeys))
+	for index, pubKey := range dkgPubKeys {
+		cdcDKGKeys[index] = cadence.NewString(pubKey.String())
+	}
+	args = append(args, cadence.NewArray(cdcDKGKeys))
+
+	return cadence.NewArray(args)
```

Comment:
Looking at the example transaction arguments file from [here](https://github.com/dapperlabs/flow-go/issues/5442#issuecomment-842718013), it looks like the outer argument list might just be a regular array rather than a cadence array. The outer array has the form `[<elements>]` as opposed to `{type: "Array", value: [<elements>]`

Reply:
Ahh yes you are right thanks